app/services/scrape.py

```diff
@@ -34,93 +30,43 @@ def __init__(self) -> None:
         self._cache: TTLCache = TTLCache(maxsize=1024, ttl=settings.TWEEPY_CACHE_TTL)
 
     @cachedmethod(cache=_cache_func, key=_make_key("get_user_by_username"))
-    def get_user_by_username(self, username) -> TwitterUser:
-        """
-        Get details of Twitter user's information from given username.
+    def get_user_by_username(self, username: str) -> tweepy.User:
```

Comment:
Why not just return our custom class `User` but switch to `tweepy.User`? I understand the single responsibility thing but isn't it much cleaner if the endpoint does not have to deal with different different models

Reply:
Which `User` class? The one in `schemas` or `models`?

Reply:
models i mean

Reply:
That means `TwitterScraper` needs to have knowledge about database representation of Twitter user, which also violates single-responsibility principle.

Reply:
What's the point of refactoring if it gets more complicated then :/

Reply:
You will see the point in the later phase of development, when technical debt get accumulated.

Reply:
The way i see it it's pretty the same
- You cut off DB knowledge -> the endpoint need more TwitterAPI knowledge (tweepy.tweet, tweepy.user)
- And ofcourse TwitterAPI has nothing to do with our endpoint also, it's suppose to be a lower layer and is only used in `services`
- So this is more like substitute DB responsibility for TwitterAPI responsibility -> No meaning what so ever.

I dont see what wrong in using the models from DB, may be other db related operations like save/update/... can be some where else to reduce the responsibility of service.

Reply:
Fixed